### ISSUE TITLE
BUG: --autocorrect ignores :enabled false and silently rewrites code (data-loss)

### ISSUE THREAD

**Issue by Cyrik** [NoahTheDuke/splint#47]
I let claude write the ticket just so it's more complete, but I verified all of the findings and the repro, so I hope you don't mind 😁 

**Version:** splint `1.24.0` (also reproduces on `1.22.0`)

**Summary**

When a rule is disabled via `:enabled false` in `.splint.edn`, a normal `splint` run correctly skips it — but `splint --autocorrect` applies it anyway. In the case below this isn't just a cosmetic issue: the disabled rule has a known false positive on reader conditionals, and `--autocorrect` silently deletes a `:clj` branch.

**Repro**

`.splint.edn`:

```clojure
{lint/if-else-nil {:enabled false}}
```

`src/example.cljc`:

```clojure
(ns example)

(defn pick [d]
  (if (zero? d)
    #?(:clj :clj-branch :default :default-branch)
    d))
```

Run:

```
bb -Sdeps '{:deps {io.github.noahtheduke/splint {:mvn/version "1.24.0"}}}' \
   -m noahtheduke.splint --autocorrect src/
```

**Expected**

No changes — `lint/if-else-nil` is disabled. (Confirmed: without `--autocorrect`, no finding is reported.)

**Actual**

```
Replacing
(if (zero? d) nil d)
 with
(when-not (zero? d) d)
in src/example.cljc:4
```

Resulting file:

```clojure
(ns example)

(defn pick [d]
  (when-not (zero? d) d))
```

Two bugs compounded:

1. `--autocorrect` doesn't honor `:enabled false`.
2. `lint/if-else-nil` reads `#?(:clj :clj-branch :default :default-branch)` as `nil`, so it sees `(if (zero? d) nil d)` and "fixes" it — silently destroying the `:clj` and `:default` branches.

Either fix would prevent the data loss, but the `:enabled false` regression is the more general issue: users who hit a false-positive rule and disable it would reasonably expect `--autocorrect` to leave that rule alone too.

**Comment by NoahTheDuke** [comment]
Hi, as per [CONTRIBUTING.md](https://github.com/NoahTheDuke/splint/blob/9cd1a1c85b86033672be79fa2aac7acb362048a1/CONTRIBUTING.md), please do not use generative AI when contributing to this project in any form. If you can't be bothered to write it, I can't be bothered to read it.
